### PR TITLE
docs: add api info about plugin helper

### PR DESCRIPTION
**Pull request by frederickfogerty**
Add info about how to consume third party GraphQL helpers

### CODE REVIEW

**Review by luqven — APPROVED** [review]
LGTM 👍🏼 

```
      // domain and secureURLToken are not required if image URLs are already imgix URLs
```

Tiny nit: does this mean I break stuff if I do include `domain` and `secureURLToken` or does it mean that it's safe to omit? I feel like the answer is I won't break stuff by including the k,v when not needed but just want to be sure that's what this means.

File: README.md [hunk 1]
```diff
@@ -893,6 +895,72 @@ export function buildFluidImageData(
...
+| resolveUrl         | `(source) => string \| undefined` | ✔        | A callback to resolve the image URL from the source data provided to the resolver. The source data is provided by the resolver of the type you are adding the image fields too.                                                                                                   |
+| resolveWidth       | `(source) => number \| undefined` |          | This callback can be used to provide the source (uncropped) width of the target image from the source data, if known. The data passed to this callback is the same as is passed to `resolveUrl`.                                                                                  |
+| resolveHeight      | `(source) => number \| undefined` |          | This callback can be used to provide the source (uncropped) height of the target image from the source data, if known. The data passed to this callback is the same as is passed to `resolveUrl`.                                                                                 |
```
Comment (angeloashmore) [inline README.md:913]: A quick note that could help is if the width/height are not known or provided, the plugin will automatically make a network request to fetch this information. Documenting this lets plugin authors know that it's not required to get this data if they don't have it, but also that if they don't have it, an extra network request per image will be made.

File: README.md [hunk 2]
```diff
@@ -893,6 +895,72 @@ export function buildFluidImageData(
...
+| allowListFields    | `Array`                           |          | This can be used to restrict the types and fields generated to avoid creating extra unnecessary types. Possible values: url, fixed, fluid, gatsbyImageData.                                                                                                                       |
+| defaultParams      | `Object`                          |          | Default imgix parameters to apply for these images.                                                                                                                                                                                                                               |
```
Comment (angeloashmore) [inline README.md:915]: Is there the ability to provide default placeholder params? I see this is left out of the main plugin options so no worries if this is out of scope.

Setting default placeholder params could help define the quality and size of the placeholder, allowing plugin authors or, by extension, app developers, to tweak generated base64 placeholder data.

**Review by angeloashmore — APPROVED** [review]
Looks great to me! Left a couple of notes, one of which is a feature request. Overall, it looks good to publish as-is, however.

**Comment by frederickfogerty** [comment]
> Tiny nit: does this mean I break stuff if I do include `domain` and `secureURLToken` or does it mean that it's safe to omit? I feel like the answer is I won't break stuff by including the k,v when not needed but just want to be sure that's what this means.

Yeah you're right. I've updated the language here

File: README.md (hunk 2 above)
Comment (frederickfogerty) [reply README.md:915]: I'll release this in a separate PR, thanks for the reminder!
